src/Common/src/TypeSystem/Ecma/EcmaType.cs

```diff
@@ -42,6 +43,19 @@ internal EcmaType(EcmaModule module, TypeDefinitionHandle handle)
 #endif
         }
 
+        public override int GetHashCode()
+        {
```

Comment:
I've added caching to MethodDesc as well as the ECMATypeDesc. Additionally, as we need to compute hashes for creation of MethodDescs in some cases, I've added a means by which its possible to precompute the hash before allocation of the object, so that we only need to compute the hash once.